<?php
/**
 * Created by PhpStorm.
 * User: zhouzzz
 * Date: 2018/5/15
 * Time: 22:54
 */

namespace app\lib\exception;


use Exception;
use think\exception\Handle;
use think\Log;
use think\Request;

class ExceptionHandle extends Handle
{
    private $code;
    private $msg;
    private $errorCode;
    //需要返回客户端当前请求的URL路径
    public function render(Exception $e)
    {
       if($e instanceof BaseException)
       {
           $this->code = $e->code;
           $this->msg = $e->msg;
           $this->errorCode = $e->errorCode;
           //如果是自定义的异常
       }else{
            if(config('app_debug'))
            {
                return parent::render($e);
                //TODO:return default error page
            }else {
                $this->code = 500;
                $this->msg = '服务器内部错误';
                $this->errorCode = 999;
                $this->recordErrorLog($e);
            }
       }
       $request = Request::instance();
       $result = [
           'msg' => $this->msg,
           'error_code' => $this->errorCode,
           'request_url' => $request->url()
       ];
       return json($result,$this->code);
//        return parent::render($e); // TODO: Change the autogenerated stub
    }

    private function recordErrorLog(Exception $e)
    {
        Log::init([
           'type' => 'File',
            'path' => LOG_PATH,
            'level' => ['error']
        ]);
        Log::record($e->getMessage(),'error');
    }
}